theopensystemslab/planx-new
# feat: Embed passport variables in GovPay metadata

## What does this PR do?
 - Imports types and helper function `formatGovPayMetadata()` from `planx-core` (PR here - https://github.com/theopensystemslab/planx-core/pull/348)
   - Removes tests as this is now covered in `planx-core`
 - Toggle input variant if value starts with "@"

**To Do**
- [x] Update `planx-core` across projects once https://github.com/theopensystemslab/planx-core/pull/348 reviewed and merged
- [x] Manually test passport variables in invite-to-pay journey on Pizza 🚀 

## How to test
 - This is a nice test flow - https://3015.planx.pizza/buckinghamshire/temporary-pay-test
 - Make payment, inspect network tab for attached metadata (see video below)
 - Make ITP payment. Network tab can't be checked as payment generated API side.
 - Both methods can be tested using the GovPay API, e.g. - 
 
```sh
curl -H "Authorization: Bearer <GOVPAY_STAGING_TOKEN>" https://publicapi.payments.service.gov.uk/v1/payments/<PAYMENT_ID>
```

(The staging token value can be reproduced in plain text using the `decrypt()` script util with the `team_integrations. staging_govpay_secret` value stored in the database. See PR https://github.com/theopensystemslab/planx-new/pull/3019 for a nicer way of calling this.)

## Demo

### Set passport variable in Editor
https://github.com/theopensystemslab/planx-new/assets/20502206/a23367c2-f3bd-4807-acbd-4f7ac37e9c9c

### Value read from passport, validated, then passed along to GovPay

https://github.com/theopensystemslab/planx-new/assets/20502206/2b71e8d6-394b-4de0-97b8-5f5eaf99b9f5
